### PR TITLE
fix: refactor referral to relationship field

### PR DESCRIPTION
**Pull request by rebelchris**
Migrated referral column to be a relationship to a user.
Also modified the `newUser` route to handle legacy input + new input.
And added 2 test cases for these scenario's

WT-988

### CODE REVIEW

File: src/entity/User.ts [hunk 1]
```diff
@@ -267,7 +276,7 @@ export const addNewUser = async (
         profileConfirmed: data.profileConfirmed,
         infoConfirmed: isInfoConfirmed(data),
         createdAt: data.createdAt,
-        referral: data.referral,
+        referralId: data.referralId,
```
Comment (idoshamun) [inline src/entity/User.ts:279]: Should it not be?

```suggestion
        referralId: data.referralId || data.referral,
```

File: src/entity/User.ts [hunk 2]
```diff
@@ -118,12 +126,13 @@ export type AddUserData = Pick<
...
   | 'github'
   | 'twitter'
-  | 'referral'
+  | 'referralId'
   | 'infoConfirmed'
   | 'profileConfirmed'
   | 'acceptedMarketing'
   | 'timezone'
 >;
+export type AddUserDataPost = { referral: string } & AddUserData;
```
Comment (idoshamun) [inline src/entity/User.ts:135]: `{ referral: string }` this part is no longer applicable right?

**Review by idoshamun — APPROVED** [review]

File: src/entity/User.ts (hunk 2 above)
Comment (rebelchris) [reply src/entity/User.ts:135]: Yep, it's just added there to retroactively support existing implementations.
We merge referral to referralId if it exists.

Comment (rebelchris) [reply src/entity/User.ts:135]: It's a new type only for the post option

**Comment by idoshamun** [comment]
Ok let's do it :)

**Pull request merged**